### PR TITLE
SDIT-1497 To Nomis sync updates

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt [hunk 1]
```diff
@@ -36,13 +40,17 @@ class LocationsService(
         eventTelemetry = telemetryMap
 
         checkMappingDoesNotExist {
-          mappingService.getMappingGivenDpsIdOrNull(event.additionalInformation.id)
+          kotlin.runCatching { mappingService.getMappingGivenDpsIdOrNull(event.additionalInformation.id) }
```
[the remark this answers is not in the document]
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt:43]: whoops, yes! Will remove

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt [hunk 2]
```diff
@@ -56,68 +64,178 @@ class LocationsService(
...
       runCatching {
-        val nomisId =
-          mappingService.getMappingGivenDpsId(dpsId).nomisLocationId
-            .also { telemetryMap += "nomisId" to it.toString() }
-
-        nomisApiService.deleteLocation(nomisId)
-        mappingService.deleteMapping(dpsId)
+        mappingService.getMappingGivenDpsId(event.additionalInformation.id).apply {
+          telemetryMap["nomisId"] = nomisLocationId.toString()
+          nomisApiService.deactivateLocation(nomisLocationId, DeactivateRequest())
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt:111]: have you considered having a single gnomes api that do all this functions? Will you have any issues if some succeed and some fail or are they all idempotent?

File: src/main/resources/application-localstack.yml [hunk 3]
```diff
@@ -30,7 +30,15 @@ hmpps.sqs:
       queueName: hmpps_prisoner_to_nomis_location_queue
       dlqName: hmpps_prisoner_to_nomis_location_dlq
       subscribeTopicId: domainevents
-      subscribeFilter: "{\"eventType\":[\"location.inside.prison.created\", \"location.inside.prison.deleted\"]}"
+      subscribeFilter: >-
```
Comment (andymarke) [inline src/main/resources/application-localstack.yml:33]: i like this notation!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt [hunk 4]
```diff
@@ -56,68 +64,178 @@ class LocationsService(
...
+      telemetryClient.trackEvent("location-delete-ignored", telemetryMap)
     }
   }
 
-  suspend fun deleteAllLocations() {
-    mappingService.getAllMappings().forEach { mapping ->
+  private suspend fun doUpdateLocation(event: LocationDomainEvent, name: String, update: suspend (Long, Location) -> Unit) {
+    val telemetryMap = mutableMapOf("dpsId" to event.additionalInformation.id)
+    telemetryMap["key"] = event.additionalInformation.key
+    if (isDpsCreated(event.additionalInformation)) {
```
[the remark this answers is not in the document]
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt:130]: Maybe a slighly misleading function name, it refers to the creation of the event being on the dps side

**Review by andymarke — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt (hunk 2 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsService.kt:111]: These are really just a placeholder as we are not yet sure how delete should be handled (except that 'real' delete is probably inadvisable due to references from other areas)

**Pull request merged**